AzureCalling/Controllers/JoinCallViewController.swift

```diff
@@ -0,0 +1,322 @@
+    var displayName: String?
+
+    private var joinCallType: JoinCallType = .groupCall
+
+    private var typeTitle: FluentUI.Label!
+    private var joinIdTextField: IconTextField!
+    private var displayNameField: IconTextField!
+    private var callTypeSelector: FluentUI.SegmentedControl!
+    private var actionButton: FluentUI.Button!
+    private var contentView: UIView!
+
+    // Handle keyboard scrolling the content
+    private var bottomConstraint: NSLayoutConstraint!
+    private var keyboardObserver: NSObjectProtocol!
+
+    // MARK: UIViewController lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Join"
+        view.backgroundColor = .white
+
+        contentView = setupUI()
+
+        keyboardObserver = NotificationCenter.default.addObserver(
+            forName: UIResponder.keyboardWillChangeFrameNotification,
+            object: nil, queue: .main,
+            using: handleKeyboardNotification(notification:)
+        )
+        // Dismiss keyboard if tapping outside
+        let tap = UITapGestureRecognizer(target: self.view, action: #selector(UIView.endEditing))
+        view.addGestureRecognizer(tap)
+    }
+
+    deinit {
+        displayNameField?.delegate = nil
+        joinIdTextField?.delegate = nil
+        NotificationCenter.default.removeObserver(keyboardObserver as Any)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        joinIdTextField.becomeFirstResponder()
+    }
+
+    // MARK: Private Functions
+    private func handleKeyboardNotification(notification: Notification) {
+        guard let userInfo = notification.userInfo,
+              let timing = userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as? NSNumber,
+              let curve = userInfo[UIResponder.keyboardAnimationCurveUserInfoKey] as? NSNumber,
+              let frameEnd = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue,
+              let window = view.window else {
+            return
+        }
+
+        let endRect = frameEnd.cgRectValue
+        let options = UIView.AnimationOptions(rawValue: curve.uintValue << 16)
+        let duration = timing.doubleValue
+
+        // Get keyboard height.
+        let keyboardFrame = view.convert(endRect, from: window)
+        let intersection = view.bounds.intersection(keyboardFrame)
+        let keyboardHeight = intersection.isNull ? 0 : intersection.size.height
+
+        UIView.animate(withDuration: duration, delay: 0, options: options) { [weak self] in
+            let bottomInset = keyboardHeight > 0 ? keyboardHeight : self?.view.safeAreaInsets.bottom
+            self?.bottomConstraint?.constant = -(bottomInset ?? 0)
+            self?.view.layoutIfNeeded()
+        }
+    }
+
+    private func validateMeetingLink() -> Bool {
+        if let joinId = joinIdTextField.text?.trimmingCharacters(in: .whitespacesAndNewlines) {
```

Comment:
https://github.com/microsoft/swift-guide/blob/gh-pages/EarlyReturn.md

Reply:
This is a good place to use a guard statement. Updating.